x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -116,601 +55,30 @@ export const signalRulesAlertType = ({
       services,
       params,
       spaceId,
-      updatedBy: updatedByUser,
+      updatedBy,
+      state,
+      name,
+      tags,
+      createdBy,
     }) {
-      const {
-        anomalyThreshold,
-        from,
-        ruleId,
-        index,
-        eventCategoryOverride,
-        filters,
-        language,
-        maxSignals,
-        meta,
-        machineLearningJobId,
-        outputIndex,
-        savedId,
-        query,
-        to,
-        threshold,
-        threatFilters,
-        threatQuery,
-        threatIndex,
-        threatIndicatorPath,
-        threatMapping,
-        threatLanguage,
-        timestampOverride,
-        type,
-        exceptionsList,
-        concurrentSearches,
-        itemsPerSearch,
-      } = params;
-
-      const searchAfterSize = Math.min(maxSignals, DEFAULT_SEARCH_AFTER_PAGE_SIZE);
-      let hasError: boolean = false;
-      let result = createSearchAfterReturnType();
-      const ruleStatusClient = ruleStatusSavedObjectsClientFactory(services.savedObjectsClient);
-      const ruleStatusService = await ruleStatusServiceFactory({
+      return ruleExecutor({
```

Comment:
Nit: since this directly returns a promise, `async` could be removed from `async executor({`